### PR TITLE
Location service

### PR DESCRIPTION
**Pull request by stellakunzang**
Add factories for testing and db seeds 
Create location service 
Add before_save to model using location service
Everything is tested

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -9,13 +9,17 @@ gem 'rails', '~> 5.1.7'
 gem 'pg'
 gem 'puma', '~> 3.7'
 gem 'graphql'
+gem 'figaro'
+gem 'faraday'
```
Comment (stellakunzang) [inline Gemfile:13]: old friends!

File: db/seeds.rb [hunk 2]
```diff
@@ -6,8 +6,25 @@
 #   movies = Movie.create([{ name: 'Star Wars' }, { name: 'Lord of the Rings' }])
 #   Character.create(name: 'Luke', movie: movies.first)
 
+user = FactoryBot.create(:user)
+15.times do
+  FactoryBot.create(:street_art, user: user)
+end
```
Comment (stellakunzang) [inline db/seeds.rb:12]: So tidy 😍

File: spec/factories/street_art.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+    longitude { Faker::Address.longitude }
+    address { Faker::Address.street_address }
+    city { Faker::Movies::LordOfTheRings.location }
+    state { Faker::Address.state_abbr }
+    zipcode { Faker::Address.zip }
+    image_urls {[ Faker::Fillmurray.image, Faker::Fillmurray.image, Faker::Fillmurray.image ]}
+    description { Faker::Movies::StarWars.quote }
+    artist_name { Faker::Artist.name }
+  end
+end
```
Comment (stellakunzang) [inline spec/factories/street_art.rb:14]: Feel free to make suggestions if these Faker choices offend you and you would rather use some different ones!

File: db/seeds.rb (hunk 2 above)
Comment (Gallup93) [reply db/seeds.rb:12]: So glad you implemented this. My tests could use some cleaning up!

**Review by Gallup93 — APPROVED** [review]

File: db/seeds.rb (hunk 2 above)
Comment (elguapogordo) [reply db/seeds.rb:12]: it won't take much time to update them now

File: app/models/street_art.rb [hunk 4]
```diff
@@ -8,4 +9,27 @@ class StreetArt < ApplicationRecord
   validates_presence_of :state
   validates_presence_of :zipcode
   validates_presence_of :image_urls
+
+  def validate_location
+    get_coordinates if latitude.nil? || longitude.nil?
+    get_address if address.nil? || zipcode.nil? || city.nil? || state.nil?
+  end
```
Comment (elguapogordo) [inline app/models/street_art.rb:16]: spiffy

File: app/services/location_service.rb [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+      req.params[:key] = ENV["MAPQUEST_API_KEY"]
+      req.params[:location] = coordinates
+    end
+    JSON.parse(response.body, symbolize_names: true)
+  end
+
+  def conn
+    Faraday.new("http://www.mapquestapi.com/")
+  end
+end
```
Comment (elguapogordo) [inline app/services/location_service.rb:22]: the api key can be moved into the conn method if you want to make it even cleaner

**Review by elguapogordo — APPROVED** [review]

**Pull request merged**